smart-on-fhir-app/src/js/oauth.js

```diff
@@ -0,0 +1,101 @@
+/* 
+  This global variable stores the application token
+  The schema for application token is
+  {
+    "access_token" : "access_token_value",
+    "expires_in"  : "validity of the token",
+    "onSetTime" : "time of token creation"
+  } 
+*/
+var application_token;
+
+// Function to perform the OAuth2.0 authorization
+// It first checks whether application token is present in the local storage
+// If present it checks whether the token has timedout or not
+// If it's still valid, then it's reused.
+// In all other cases, a new token is fetched.
+function oauth() {
+  // Get the token from local storage
+  application_token = window.localStorage.getItem("application_token");
+
+  if (application_token == null) {
+    // If token is not found
+    fetchNewAccessToken();
+  } else {
+    application_token = JSON.parse(application_token);
+    // Check whether the token has timed-out
+    if (checkTimeout()) {
+      fetchNewAccessToken();
+    }
+  }
+}
+
+// To check whether the token has timed-out or not
+function checkTimeout() {
+  let timeOutTime = application_token["expires_in"];
+  let onSetTime = application_token["onSetTime"];
+  let currentTime = new Date().getTime();
+
+  if (currentTime - onSetTime > timeOutTime) {
+    return true;
+  }
+  return false;
+}
+
+// The function fetches a new access token.
+// It uses the client_id, client_secret and oauth_url
+// stored in the local storage
+function fetchNewAccessToken() {
+  let client_id = window.localStorage.getItem("client_id");
+  let client_secret = window.localStorage.getItem("client_secret");
+  let oauth_url = window.localStorage.getItem("oauth_url");
+
+  let auth_Headers = new Headers();
+  auth_Headers.append("Content-Type", "application/json");
+
+  let raw =
```

Comment:
use string temples/ object here. If using object, find a way to make it to string.